homeassistant/components/apple_tv/__init__.py

```diff
@@ -103,26 +87,19 @@ async def async_migrate_entry(hass, config_entry):
     """Migrate old entry."""
     _LOGGER.debug("Migrating from version %s", config_entry.version)
 
-    if config_entry.version == 1:
+    if config_entry.version in [1, 2]:
         new = {**config_entry.data}
 
-        # Not used anymore
-        del new[CONF_PROTOCOL]
+        new.setdefault(CONF_IDENTIFIERS, []).append(config_entry.unique_id)
 
         config_entry.data = {**new}
-        config_entry.version = 2
+        config_entry.version = 3
```

Comment:
We shouldn't change config entry version or use the migration api. Just check for missing item in `async_setup_entry` and update the config entry data etc as needed.

Reply:
Huh? Is it deprecated? I thought the purpose of it was to ensure version and code matches to avoid having to make optional checks.

Reply:
As I mentioned above, we want to avoid making rollback impossible. Increasing the config entry version will make rolling back impossible.

Reply:
Ok, sure, it wasn't very clear from the documentation. Then I guess I can remove the migrator completely then.

Reply:
I changed back to version 1, removed the migration method and added a check if `CONF_IDENTIFIERS` is present in the config entry during setup (and add it if it's not).